src/ConstructionBase.jl

```diff
@@ -55,6 +57,16 @@ function setproperties(obj; kw...)
 end
 
 setproperties(obj, patch::NamedTuple) = _setproperties(obj, patch)
+setproperties(obj::Tuple, patch::Tuple) = patch # TODO: should we check lengths?
```

Comment:
* Should `setproperties((10,20,30), (100,200))` be allowed? I think so, since `propertynames(obj) = (1,2,3)` contains `propertynames(patch) = (1,2)`. 
* Should `setproperties((10,20), (100,200,300))` be allowed? I think not since `propertynames(obj)` does not contain `propertynames(patch)`